client/src/App.tsx

```diff
@@ -8,85 +8,194 @@ import {
   Form,
   FormGroup,
   Input,
+  FormFeedback,
+  Spinner,
 } from "reactstrap";
 import { Chart } from "react-charts";
 import useDemoConfig from "./styles/useDemoConfig";
 
-const dataFromApiAdvanced = [
-  {
-    n: 7,
-    coolingFactor: [
-      { x: 0, y: 21 },
-      { x: 10, y: 20 },
-      { x: 20, y: 20 },
-      { x: 30, y: 30 },
-      { x: 40, y: 60 },
-      { x: 50, y: 100 },
-      { x: 60, y: 150 },
-      { x: 70, y: 170 },
-      { x: 80, y: 180 },
-      { x: 90, y: 185 },
-    ],
-    temperature: [
-      { x: 0, y: 21 },
-      { x: 10, y: 20 },
-      { x: 20, y: 20 },
-      { x: 30, y: 30 },
-      { x: 40, y: 60 },
-      { x: 50, y: 100 },
-      { x: 60, y: 150 },
-      { x: 70, y: 170 },
-      { x: 80, y: 180 },
-      { x: 90, y: 185 },
-    ],
-  },
-  {
-    n: 8,
-    coolingFactor: [
-      { x: 0, y: 21 },
-      { x: 10, y: 20 },
-      { x: 20, y: 20 },
-      { x: 30, y: 30 },
-      { x: 40, y: 60 },
-      { x: 50, y: 100 },
-      { x: 60, y: 150 },
-      { x: 70, y: 170 },
-      { x: 80, y: 180 },
-      { x: 90, y: 185 },
-    ],
-    temperature: [
-      { x: 0, y: 21 },
-      { x: 10, y: 20 },
-      { x: 20, y: 20 },
-      { x: 30, y: 30 },
-      { x: 40, y: 60 },
-      { x: 50, y: 100 },
-      { x: 60, y: 150 },
-      { x: 70, y: 170 },
-      { x: 80, y: 180 },
-      { x: 90, y: 185 },
-    ],
-  },
-];
-
-const dataFromApi = [
-  { x: 0, y: 21 },
-  { x: 10, y: 20 },
-  { x: 20, y: 20 },
-  { x: 30, y: 30 },
-  { x: 40, y: 60 },
-  { x: 50, y: 100 },
-  { x: 60, y: 150 },
-  { x: 70, y: 170 },
-  { x: 80, y: 180 },
-  { x: 90, y: 185 },
-];
+const isNumber = (num?: number): num is number => {
+  return Number(num) === 0 ? true : !!Number(num);
+};
+
+// const dataFromApiAdvanced = [
```

Comment:
I've decided to keep these arrays to better understand the shape of the upcoming data from the server api and to also test the client side without the server running.